tests/Unit/Helpers/Elliptic/FirstOrderSystem.hpp

```diff
@@ -0,0 +1,279 @@
+  using argument_tags = typename SourcesComputer::argument_tags;
+
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> dist(-1., 1.);
+
+  // Generate random variables and fluxes
+  const auto vars = make_with_random_values<VarsType>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto fluxes = make_with_random_values<FluxesType>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+
+  // Generate sources from the variables with random arguments
+  tuples::tagged_tuple_from_typelist<argument_tags> args{};
+  tmpl::for_each<argument_tags>(
+      [&args, &generator, &dist, &used_for_size](auto tag_v) {
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        get<tag>(args) = make_with_random_values<typename tag::type>(
+            make_not_null(&generator), make_not_null(&dist), used_for_size);
+      });
+  const auto computed_sources = tuples::apply(
+      [&vars, &fluxes](const auto&... expanded_args) {
+        return ::elliptic::first_order_sources<
+            volume_dim, primal_fields, auxiliary_fields, SourcesComputer>(
+            vars, fluxes, expanded_args...);
+      },
+      args);
+
+  // Create a DataBox
+  auto box = tuples::apply(
+      [&vars, &fluxes, &used_for_size](const auto&... expanded_args) {
+        return db::create<tmpl::append<
+            tmpl::list<vars_tag, fluxes_tag, sources_tag>, argument_tags>>(
+            vars, fluxes,
+            make_with_value<typename sources_tag::type>(
+                used_for_size, std::numeric_limits<double>::signaling_NaN()),
+            expanded_args...);
+      },
+      args);
+
+  // Apply the sources computer to the DataBox
+  tuples::tagged_tuple_from_typelist<
+      tmpl::append<primal_fields,
+                   db::wrap_tags_in<::Tags::Flux, primal_fields,
+                                    tmpl::size_t<volume_dim>, Frame::Inertial>>>
+      vars_tuple{};
+  tmpl::for_each<primal_fields>([&vars, &fluxes, &vars_tuple](auto tag_v) {
+    using tag = tmpl::type_from<decltype(tag_v)>;
+    using flux_tag =
+        ::Tags::Flux<tag, tmpl::size_t<volume_dim>, Frame::Inertial>;
+    get<tag>(vars_tuple) = get<tag>(vars);
+    get<flux_tag>(vars_tuple) = get<flux_tag>(fluxes);
+  });
+  tmpl::for_each<auxiliary_fields>([&vars, &box](auto tag_v) {
+    using tag = tmpl::type_from<decltype(tag_v)>;
+    db::mutate<::Tags::Source<tag>>(
+        make_not_null(&box),
+        [&vars](const auto aux_source) { *aux_source = get<tag>(vars); });
+  });
+  tuples::apply(
+      [&box](const auto&... expanded_vars) {
+        db::mutate_apply<
+            db::wrap_tags_in<::Tags::Source, typename vars_tag::tags_list>,
+            argument_tags>(SourcesComputer{}, make_not_null(&box),
+                           expanded_vars...);
+      },
+      vars_tuple);
+  CHECK(computed_sources == get<sources_tag>(box));
+}
+
+template <typename Tag>
```

Comment:
could these tags be in a `detail` namespace? It doesn't look like these are meant to be part of the interface